Orange/widgets/widget.py

```diff
@@ -738,6 +745,32 @@ def _userconfirmed():
 
         self.__msgwidget.accepted.connect(_userconfirmed)
 
+    @classmethod
+    def migrate_settings(cls, settings, version):
```

Comment:
> The widget is therefore responsible for migration of all the settings.

Say that some component is reused in multiple widgets. If we change a setting in the component, we have to add the corresponding migration to all widgets? Including, say, widgets in third-party add-ons that use our components? With the later, any change in component's settings would break compatibility with widgets; with introducing component-level migrations, this can be avoided.

Wouldn't it thus be better if settings handler traversed all components (including the widget itself) and called the corresponding migration method in each? This would also require setting versioning on components, not widgets.